lib/bitcoin/BitcoinProcessor.ts

```diff
@@ -176,8 +188,130 @@ export default class BitcoinProcessor {
+   * @param startingBlock the starting block to begin processing
+   */
+  private async fastProcessTransactions (startingBlock: IBlockInfo) {
+    const fileReader = new BitcoinFileReader(this.bitcoinDataDirectory!);
+    const blockFiles = fileReader.listBlockFiles().sort();
+    let fileIndex = blockFiles.length - 1;
+
+    const bestHeight = await this.bitcoinClient.getCurrentBlockHeight();
+    const bestHash = await this.bitcoinClient.getBlockHash(bestHeight);
+
+    let currentHash = bestHash;
+    let previousHash = undefined;
+    let numOfBlocksToProcess = bestHeight - startingBlock.height + 1;
+
+    const feeInfos = new Array(numOfBlocksToProcess).fill(undefined);
+
+    console.log(`Begin fast processing block ${startingBlock.height} to ${bestHeight}`);
+    // loop through files and process them until we process all blocks needed
+    while (numOfBlocksToProcess !== 0 && fileIndex >= 0) {
+      const blockFileName = blockFiles[fileIndex];
+      fileIndex--;
+      const blockFile = fileReader.readBlockFile(blockFileName);
+      const blockData = BitcoinRawDataParser.parseRawDataFile(blockFile);
+      // for each block within the current file, add fee info and write to db
+      for (let key in blockData) {
+        const block = blockData[key];
+
+        const indexInFeeInfo = block.height - startingBlock.height;
+        if (indexInFeeInfo >= 0 && indexInFeeInfo < numOfBlocksToProcess) {
+          this.addBlockToFeeInfos(feeInfos, indexInFeeInfo, block);
+          await this.processSidetreeTransactionsInBlock(block);
+        }
+      }
+
+      // look through fee info to see if there are confirmed data
+      // we only need to look from numOfBlocksToProcess because we know the elements after it are confirmed
+      let currentFeeData = feeInfos[numOfBlocksToProcess - 1];
+      while (currentFeeData !== undefined && currentFeeData[currentHash] !== undefined) {
+        // delete all unneeded key value pairs and db records
+        const hashes = Object.keys(currentFeeData);
+        for (let hash of hashes) {
+          if (hash !== currentHash) {
+            delete currentFeeData[hash];
+            await this.transactionStore.removeTransactionByTransactionTimeHash(hash);
+          }
+        }
+        const confirmedFeeData = currentFeeData[currentHash];
+        if (currentHash === bestHash) {
+          previousHash = confirmedFeeData.prevHash;
+        }
+        currentHash = confirmedFeeData.prevHash;
+        numOfBlocksToProcess--;
+        currentFeeData = feeInfos[numOfBlocksToProcess - 1];
+      }
+    }
+
+    this.lastProcessedBlock = {
+      height: bestHeight,
+      hash: bestHash,
+      previousHash: previousHash
+    };
+    console.log('finished fast processing');
+    // TODO: Issue #783
+    // Need to use the fee infos and set fee after here.
+  }
+
+  private addBlockToFeeInfos (feeInfos: any[], indexInFeeInfo: number, block: BitcoinBlockModel) {
```

Comment:
Generally speaking, I'm uneasy about `any` types. We are using typescript so let's use the type-safety as much as we can IMO. Can we have an actual object type here (even an internal model to keep it within this class?)